### DISCUSSION TITLE
Fine-tuned model and inference parameters

### DISCUSSION THREAD

**Discussion by lightyear12** [fcakyon/small-object-detection-benchmark#9] [Q&A] [answered]
Hello, congratulations on the success of your project. I am trying to reproduce the experiments, I want to use tood to fine-tune on visdrone dataset, so I use the codes from the following path:
 [https://github.com/fcakyon/sahi-benchmark/blob/main/mmdet_tools/train.py](url)
[https://github.com/fcakyon/sahi-benchmark/blob/main/mmdet_configs/visdrone_tood/tood_crop_480_960_cls_60.py](url)
Due to limited server resources，we can only run batch_multiplier= 2 in the fine-tuning stage, and when we use this pth to inference on the test-dev dataset, There is a big gap between our AP results and those in the paper. 
Can you provide the tood pre-trained model(pth) on the visdrone dataset?
Looking forward to your reply.
Thank you in advance.
@fcakyon

**Comment by fcakyon** [comment]
@lightyear12 I have just uploaded coco formatted result json files, classwise evaluation results and classwise error analysis plots for all of the visdrone experiments given in paper: https://github.com/fcakyon/sahi-benchmark/blob/main/README.md#visdrone-results

Is should be enough for you if your intention is to use the results in your paper. If you need anything else, feel free to write it here 👍

**Comment by lightyear12** [comment]
Thanks for your reply！👍
But we still have some questions……
We used TOOD network Finetune to complete 24 epochs on Visdrone, and our results are shown below，I want to know whether these indicators are similar to the results of your finetune？
![e405aaf7b30b593825f7b2bb3ffdf8a](https://user-images.githubusercontent.com/96284098/162354329-64f54116-bfe5-4018-aa42-89241ed8bdfd.png)
 Later, sahi library was used to complete the inference stage, but the indicators in table1 were not available. We want to know whether the inference stage in the paper is different from the predict function in sahi library (for example, there are some pre-processing for slice)？
@fcakyon

**Comment by fcakyon** [comment]
@lightyear12 inference step is similar to one in [sahi](https://github.com/obss/sahi) library. If you specifically tell me which setting in the paper you are talking about, I can tell you which parameters you should use during inference. Also I can try to upload the weight file corresponding to the setting you are referring.

**Comment by lightyear12** [comment]
We want to reproduce the paper’s table1 TOOD results (Mean average precision values calculated on Visdrone19-Detection test-dev set.)
First, we need to obtain the TOOD model fine-tuned on Visdrone dataset, so we use the codes from the following path:
https://github.com/fcakyon/sahi-benchmark/blob/main/mmdet_tools/train.py
https://github.com/fcakyon/sahi-benchmark/blob/main/mmdet_configs/visdrone_tood/tood_crop_480_960_cls_60.py
we only change BATCH_MULTIPLIER in tood_crop_480_960_cls_60.py (from 8 to 2) in fine-tuning stage to obtain the weights file. Then we used sahi predict to load weights file to inference. The relevant parameters are:
slice_height = 256
slice_width = 256
overlap_height_ratio = 0.2
overlap_width_ratio = 0.2

1. Could you tell us which parameters we should use during inference?
2. Could you upload the TOOD weights file fine-tuned on Visdrone dataset?
@fcakyon

**Comment by fcakyon** [comment, marked as answer]
@lightyear12 here is the important inference parameters detailed in paper:

    model_confidence_threshold=0.01
    no_standard_prediction=False
    no_sliced_prediction=False
    slice_height=640
    slice_width=640
    overlap_height_ratio=0.25
    overlap_width_ratio=0.25
    postprocess_type="NMS"
    postprocess_match_metric="IOU"

Will upload the weights as soon as I can (atm busy with deadlines) 👍

**Comment by lightyear12** [comment]
Thanks for your reply, we will do our best to keep trying! @fcakyon

**Comment by bshakya77** [comment]
Hello @fcakyon,

I see hyperparameter values are static in current sahi framework like `overlap_height_ratio=0.25 ` and `overlap_width_ratio=0.25`. Is the current framework designed for the static value only? Can it be modified to take dynamic values to reduce slicing? 

Thank you.

Regards,
Bijay
